secret/secrets_manager_vault.go

```diff
@@ -20,20 +22,36 @@ func NewBasicSecretsManager(c SecretsManagerClient) *BasicSecretsManager {
 
 // CreateSecret creates a new secret.
 func (m *BasicSecretsManager) CreateSecret(ctx context.Context, s NamedSecret) (id string, err error) {
-	return "", errors.New("TODO: implement")
+	out, err := m.client.CreateSecret(ctx, &secretsmanager.CreateSecretInput{
+		Name:         s.Name,
+		SecretString: s.Value,
+	})
+	return *out.ARN, err
```

Comment:
This can panic if `out.ARN` is nil (e.g. if `err` is non-nil, expect `out.ARN` to be nil). Check `if err != nil` before returning.
Also, AWS has a bad habit of passing back valid output and no error, but not populating some fields. In those cases, we have to check for nil before we can dereference, so `out` might be non-nil but `out.ARN` might be nil.

As a general rule of thumb, the deference (`*`) operator should only be used if you're completely certain that it's non-nil and definitely won't panic.